core/src/main/java/org/springframework/security/core/AuthenticationException.java

```diff
@@ -46,6 +49,12 @@ public AuthenticationException(String msg, Throwable cause) {
 	 */
 	public AuthenticationException(String msg) {
 		super(msg);
+		this.authRequest = null;
+	}
+
+	public AuthenticationException(String msg, Authentication authRequest) {
```

Comment:
Since this is [an optional property](https://github.com/spring-projects/spring-security/issues/16444#issuecomment-2605931950), please add this as a setter. This also prevents a multiplicity of constructors.

Reply:
Also please [add `@since 6.5`](https://github.com/spring-projects/spring-security/blob/main/CONTRIBUTING.adoc?plain=1#L105) new public API components